src/domains/disjointDomain.ml

```diff
@@ -176,6 +178,32 @@ struct
   let disjoint m1 m2 = is_empty (inter m1 m2) (* TODO: optimize? *)
 end
 
+module type MayEqualSetDomain =
+sig
+  include SetDomain.S
+  val may_be_equal: t -> t -> bool
+end
+
+module ProjectiveSetPairwiseMeet (E: Printable.S) (B: MayEqualSetDomain with type elt = E.t) (R: Representative with type elt = E.t): SetDomain.S with type elt = E.t = struct
+  include ProjectiveSet (E) (B) (R)
+
+  let meet m1 m2 =
+    let inner_fold key b key2 b2 acc =
+      let may_be_equal = B.may_be_equal b b2 in
+      if may_be_equal then
+        acc
+        |> M.add key b
+        |> M.add key2 b2
```

Comment:
Currently, a the set abstraction used is a `Joined.t`. Now, when one has one such `Joined.t` abstraction

Reply:
Now, say there are two address sets representing `a.fst` and `a[def_exc:0][def_exc:0]`, respectively. Then, when one does a meet on these by meeting the buckets (which are currently represented by `Joined.t`), then this will right now throw an `Uncomparable` exception. The question would then in that case where currently the exception happens:
- One could just add both buckets to the resulting map, but then, without checking whether they `may_be_equal`, it is not checked that the `meet` actually needs to be non-empty. Thus, in any case when there are addresses with incomparable offsets, but the same base variable, they would be in the `meet`.  
- One could alternatively keep multiple elements at the level of the `SetDomain` used by the `ProjectiveSet`, as long as they may have the same numeric offset (but differing offset expressions). I am not certain whether we would like to have a single `SetDomain.t` track multiple offsets in this setting?

Reply:
@sim642 Any further opinion on this? I think there is no reasonable way to get around having this `may_be_equal` in one form or the other.

Reply:
The point about not using `B.meet` is right, I suppose. I simplified this implementation though, such that it doesn't construct the intermediate sets.